app/pkg/protocol/init_proposal.go

```diff
@@ -10,28 +10,29 @@ import (
 
 // InitProposal represents information about initial state, contract, participants.
 type InitProposal struct {
-	Participants []Participant
+	Participants []*Participant
```

Comment:
due to benchmark results, its more relevant to use slice of pointers to Structs
- struct is big
- as we can have big amount of participants in channel, performance of append will be much more better

Reply:
Max number of participants of a channel is 256.

So maybe we can use this limitation to have better a better way to identify participant, like using `u8` (in maps for examples, instead of using the pointer address, which is ok in the current code base, but which is kind of bad practice in general).